Sources/Views/MessageLabel.swift

```diff
@@ -22,417 +22,491 @@
+            if !isConfiguring { setNeedsDisplay() }
         }
     }
-
+    
+    public var dateAttributes: [NSAttributedStringKey: Any] = [:] {
+        didSet {
+            updateDetectorAttributes(for: .date)
+            if !isConfiguring { setNeedsDisplay() }
+        }
+    }
+    
+    // MARK: - Properties [Private]
+    
+    private var layoutManager: NSLayoutManager
+    
+    private var textContainer: NSTextContainer
+    
+    private var textStorage: NSTextStorage
+    
+    private var phoneResults: [NSRange: String] = [:]
+    
+    private var addressResults: [NSRange: [NSTextCheckingKey: String]] = [:]
+    
+    private var dateResults: [NSRange: Date] = [:]
+    
+    private var urlResults: [NSRange: URL] = [:]
+    
+    private var isConfiguring: Bool = false
+    
+    func paragraphStyle(from string: NSAttributedString) -> NSParagraphStyle {
+        
+        guard string.length > 0 else { return NSParagraphStyle() }
+        
+        var range = NSRange(location: 0, length: string.length)
+        let existingStyle = string.attribute(.paragraphStyle, at: 0, effectiveRange: &range) as? NSMutableParagraphStyle
+        let style = existingStyle ?? NSMutableParagraphStyle()
+        
+        style.lineBreakMode = lineBreakMode
+        style.alignment = textAlignment
+
+        return style
+        
+    }
+    
     // MARK: - Initializers
-
+    
     public override init(frame: CGRect) {
+        
+        self.textContainer = NSTextContainer()
+        self.textContainer.lineFragmentPadding = lineFragmentPadding
+        self.textContainer.maximumNumberOfLines = numberOfLines
+        self.textContainer.lineBreakMode = lineBreakMode
+        self.textContainer.size = frame.size
+        
+        self.layoutManager = NSLayoutManager()
+        self.layoutManager.addTextContainer(self.textContainer)
+        
+        self.textStorage = NSTextStorage(attributedString: NSAttributedString(string: ""))
+        textStorage.addLayoutManager(layoutManager)
+        
         super.init(frame: frame)
-
-        // Message Label Specific
-        self.numberOfLines = 0
-        self.lineBreakMode = .byWordWrapping
-
+        
         let defaultAttributes: [NSAttributedStringKey: Any] = [
-          NSAttributedStringKey.foregroundColor: self.textColor,
-          NSAttributedStringKey.underlineStyle: NSUnderlineStyle.styleSingle.rawValue,
-          NSAttributedStringKey.underlineColor: self.textColor
+            .underlineStyle: NSUnderlineStyle.styleSingle.rawValue,
+            .underlineColor: UIColor.darkText
         ]
-
+        
         self.addressAttributes = defaultAttributes
+        self.phoneAttributes = defaultAttributes
         self.dateAttributes = defaultAttributes
-        self.phoneNumberAttributes = defaultAttributes
         self.urlAttributes = defaultAttributes
-
+        
         setupGestureRecognizers()
-
+        
+        backgroundColor = .clear
```

Comment:
I think by default the background of a UIView is already nil (clear)

Reply:
For some reason it was `UIColor.black`

Reply:
👍